### ISSUE TITLE
[MA0151] False warning when use `typeof` keyword

### ISSUE THREAD

**Issue by CorrM** [meziantou/Meziantou.Analyzer#716]
### Version of the Meziantou.Analyzer NuGet package

2.0.149

### Rule Identifier

MA0151

### Target Framework

net8.0

### C# Language version

C# 12

### Description

DebuggerDisplay in code below show `MA0151: Member 'typeof' does not exist`.

### Reproduction Steps

Minimal code:

```c#
[Serializable]
    [DebuggerDisplay("{" + nameof(Pointer) + "}, Type = {typeof(T).Name}")]
    public sealed class FPointer<T>
    {
        public required nuint Pointer { get; init; }

        public bool IsNotNull()
        {
            return Pointer.IsNotNull();
        }

        public bool IsNull()
        {
            return Pointer.IsNull();
        }
    }
```

**Comment by meziantou** [comment]
Sorry for the late reply

Any expression that a debugger can evaluate is valid in the `[DebuggerDisplay]`. However, it's not possible for an analyzer to validate it correctly. For instance, your expression may not be valid if someone uses your library from a VB.NET project.

A recommended pattern in .NET is to create a private property to get the debugger value:

````
[DebuggerDisplay("{DebuggerDisplay,nq}")]
class Sample<T>
{
    private string DebuggerDisplay => typeof(T).Name;
}
````

This has multiple advantages:
- You can rely on the compiler to validate the content of the expression as this is now code
- It works in C#, VB.NET, F#, etc.
